activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -104,6 +104,13 @@ def initialize(connection, logger, connection_options, config)
         @active     = true
         @statements = StatementPool.new(self.class.type_cast_config_to_integer(config[:statement_limit]))
 
+        if sqlite_version < "3.8.0"
+          $stderr.puts <<-EOS.squish
+            "Your version of SQLite (#{sqlite_version.to_s}) is too old. Active Record supports SQLite >= 3.8."
+          EOS
+          exit 1
```

Comment:
Is there any reason not to use `raise` unlike other adapters?

https://github.com/rails/rails/blob/ce4d467f7c2fc812e257a87bd4875c1f1f08a981/activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb#L56-L58

https://github.com/rails/rails/blob/ce4d467f7c2fc812e257a87bd4875c1f1f08a981/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L220-L222

Reply:
I'm thinking about implementing version check like rails/rails#28939 just showing a necessary message for users by omitting stack traces like this.

```ruby
$ sqlite3 --version
3.7.9 2011-11-01 00:52:41 c7c6050ef060877ebe77b41d959e9df13f8c9b5e
```
```ruby
$ bundle exec ruby -w -Itest test/cases/adapters/sqlite3/sqlite3_adapter_test.rb
Using sqlite3
"Your version of SQLite (3.7.9) is too old. Active Record supports SQLite >= 3.8."
$
```
I'd like to get some feedback on this change. I agree to make all bundled adapters behave similarly for the database version check. then I'll revert this change or open another pull request to change mysql2 and posgresql adapter version checks.

Reply:
Usually, a library does/should not terminate a process itself.
If a library may terminate a process itself, it hard to handle that case by users.

Reply:
Understood. Thanks for the comment.